test_scripts/Polices/External_UCS/030_ATF_N_TC_LPT_Update_PreloadedPT_with_disallowed_by_external_consent_entities_on_struct_with_invalid_type_of_params.lua

```diff
@@ -0,0 +1,193 @@
+-- c. shut SDL down
+--
+-- Preconditions:
+-- 1. Stop SDL (Ignition Off)
+-- 2. Modify PreloadedPolicyTable (add 'disallowed_by_external_consent_entities_on' section)
+-- Define not valid data type for entityType and entityId parameters
+-- 3. Initiate Local Policy Table update by setting 'preloaded_date' parameter
+--
+-- Steps:
+-- 1. Start SDL (Ignition On)
+-- 2. Check SDL status
+--
+-- Expected result:
+-- Status = 0 (SDL is stopped)
+--
+-- Note: Script is designed for EXTERNAL_PROPRIETARY flow
+---------------------------------------------------------------------------------------------
+
+--[[ General configuration parameters ]]
+config.deviceMAC = "12ca17b49af2289436f303e0166030a21e525d266e209267433801a8fd4071a0"
+config.defaultProtocolVersion = 2
+config.ExitOnCrash = false
+
+--[[ Required Shared Libraries ]]
+local commonFunctions = require('user_modules/shared_testcases/commonFunctions')
+local commonSteps = require('user_modules/shared_testcases/commonSteps')
+local commonPreconditions = require('user_modules/shared_testcases/commonPreconditions')
+local sdl = require('SDL')
+local testCasesForExternalUCS = require('user_modules/shared_testcases/testCasesForExternalUCS')
+local testCasesForPolicySDLErrorsStops = require('user_modules/shared_testcases/testCasesForPolicySDLErrorsStops')
+
+--[[ Local variables ]]
+local grpId = "Location-1"
+local checkedSection = "disallowed_by_external_consent_entities_on"
+local params = { "entityId", "entityType" }
+local values = {
+  { v = "1", desc = "String" },
+  { v = 1.23, desc = "Float" },
+  { v = {}, desc = "Empty table" },
+  { v = { entityType = 1, entityID = 1 }, desc = "Non-empty table" },
+  { v = -1, desc = "Below Min" },
+  { v = 129, desc = "Beyond Max" },
+  { v = "", desc = "Blank" },
+  { v = nil, desc = "Null" }
+}
+
+--[[ General Precondition before ATF start ]]
+commonFunctions:SDLForceStop()
+commonSteps:DeleteLogsFileAndPolicyTable()
+commonPreconditions:BackupFile("sdl_preloaded_pt.json")
+
+--[[ General Settings for configuration ]]
+Test = require("user_modules/connecttest_resumption")
+require('user_modules/AppTypes')
+
+--[[ Preconditions ]]
+commonFunctions:newTestCasesGroup("Preconditions")
+
+function Test:CheckSDLStatus()
+  testCasesForExternalUCS.checkSDLStatus(self, sdl.RUNNING)
+end
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Test")
+
+local n = 1
+local startLine = 1
+
+local function sequence(desc, updateFunc)
+
+  local tcName = "TC[" .. string.format("%02d", n) .. "] - [" .. desc .. "] "
+  tcName = tcName .. string.rep("-", 50 - string.len(tcName))
+
+  Test[tcName] = function() end
+
+  function Test:StopSDL_IGNITION_OFF()
+    testCasesForExternalUCS.ignitionOff(self)
+  end
+
+  function Test:CheckSDLStatus()
+    testCasesForExternalUCS.checkSDLStatus(self, sdl.STOPPED)
+  end
+
+  function Test.UpdatePreloadedPT()
+    testCasesForExternalUCS.updatePreloadedPT(updateFunc)
+  end
+
+  function Test.StartSDL()
```

Comment:
@dboltovskyi, please note that there is an ATF defect: "[ATF] ATF hangs-up when call SDL:StartSDL and SDL is already running.".
Please make additional action for Test:CheckSDLStatus() in case SDL is not stopped to stop it.
Please make the correction in all applicable scripts, I will also check for this remark in the scripts.